src/core/symbology-ng/qgsgeometrygeneratorsymbollayerv2.cpp

```diff
@@ -186,17 +186,22 @@ bool QgsGeometryGeneratorSymbolLayerV2::isCompatibleWithSymbol( QgsSymbolV2* sym
   Q_UNUSED( symbol )
   return true;
 }
-
 void QgsGeometryGeneratorSymbolLayerV2::render( QgsSymbolV2RenderContext& context )
 {
-  QgsGeometry geom = mExpression->evaluate( &context.renderContext().expressionContext() ).value<QgsGeometry>();
   if ( context.feature() )
   {
-    QgsFeature f = *context.feature();
+    QgsExpressionContext& expressionContext = context.renderContext().expressionContext();
+
+    QgsFeature f = expressionContext.feature();
+    QgsGeometry geom = mExpression->evaluate( &expressionContext ).value<QgsGeometry>();
     f.setGeometry( geom );
 
+    QgsExpressionContextScope* subSymbolExpressionContextScope = mSymbol->symbolRenderContext()->expressionContextScope();
+
+    subSymbolExpressionContextScope->setFeature( f );
```

Comment:
This shouldn't be needed - if the subSymbolExpressionContextScope does not have a feature set, expressionContext will fall back to using the already set feature (which must have already been done, or line 195 would not work).